### PR TITLE
[feat] Create tips tab and add loading states

### LINKED ISSUE #100: Create tips tab in resources page
https://linear.app/trap-garden/issue/TG-63/create-the-tips-tab-in-resources

### PR DESCRIPTION
**Pull request by torinnayak1**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Created the tips tab, with dropdowns that pull from the supabase.
Added loading states to view plants and planting timeline

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Screenshot 2025-03-15 at 3 55 37 PM
Screenshot 2025-03-15 at 3 55 55 PM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Maybe review the organization of TipsDisplay first (in the app/resources page) and then the construction of the Tips Component? Sorry I do have negative margins and whatnot, please lmk how to fix

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Fixing colors of the icons (making new icons perhaps)

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @kylezryr

### CODE REVIEW

File: app/resources/page.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    //Map category values to icon name
+    const categoryIcons: Record<string, IconType> = {
+      'Helpful Flowers for Your Garden': 'flower',
+      'Water Management': 'wateringCan',
+      Mulching: 'lawnCare',
+      Harvesting: 'harvestingBasket',
+      Planting: 'plantHand',
+      Weeding: 'spade',
+    };
```
Comment (kylezryr) [inline app/resources/page.tsx]: this (and `categoryHeaders`) above may be better to move into utils/helpers and exported from there for more reusability. could we also rename them to `tipCategoryIcons` and `tipCategoryHeaders`?

File: app/resources/page.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    const [fullList, setFullList] = useState<PlantTip[]>([]);
```
Comment (kylezryr) [inline app/resources/page.tsx]: would be better to rename this to something more descriptive like `fullTipList`!

File: app/resources/page.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    useEffect(() => {
+      (async () => {
+        const tipList = await getAllPlantTips();
+        setFullList(tipList);
+      })();
+    });
```
Comment (kylezryr) [inline app/resources/page.tsx]: we should add an empty dependency array here in the useEffect so that it only triggers on the first load, otherwise it continuously runs

File: app/resources/page.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    return (
+      <>
+        <Box $ml="1.5rem" $mb=".5rem">
+          <H4 $fontWeight={500} $color="#1F5A2A">
```
Comment (kylezryr) [inline app/resources/page.tsx]: use COLORS.shrub for this!

File: app/view-plants/page.tsx
Comment (kylezryr) [inline app/view-plants/page.tsx:1]: oh umm looks like the loading state changes are in this PR too... did you make a branch for those changes? if not its okay, we can include the changes in this PR and just update the PR title/description but in the future it would be best t branch for each sprint

File: app/view-plants/page.tsx [hunk 5]
```diff
@@ -272,13 +274,19 @@ export default function Page() {
...
+              handleClick={() => {
+                setViewingOption('all');
+              }}
+              buttonText="Add Plants"
+            />
+          ) : (
+            <Flex $justify="center" $align="center" $h="30rem">
+              <H2>Loading...</H2>
+            </Flex>
+          )
```
Comment (kylezryr) [inline app/view-plants/page.tsx:289]: loading state changes look good otherwise!

File: components/TipDropdown/index.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+          expanded ? 'var(--white-grey, #F0F0F0)' : 'var(--white, #FFFFFF)'
+        }
+      >
+        <Flex $gap="1rem" $direction="row" $align="center">
+          <StyledIcon type={icon} />
+          <Flex $direction="column" $gap=".25rem">
+            <CategoryName $colorString={expanded ? COLORS.shrub : 'black'}>
+              {name}
+            </CategoryName>
+            <P3 $color="var(--medium-grey, #888)">{tips.length} Tips</P3>
```
Comment (kylezryr) [inline components/TipDropdown/index.tsx]: we can use COLORS.midgray for #888

File: components/TipDropdown/index.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+              height="6"
+              viewBox="0 0 11 6"
+              fill="none"
+              transform="rotate(180)"
+            >
+              <path
+                d="M5.85387 6C6.08883 6 6.28367 5.91404 6.46705 5.73066L10.7822 1.31805C10.9312 1.16905 11 0.991404 11 0.77937C11 0.34957 10.6619 0 10.2321 0C10.0201 0 9.82521 0.0916905 9.67049 0.246418L5.8596 4.16619L2.03725 0.246418C1.88252 0.0916905 1.69341 0 1.47565 0C1.04585 0 0.702006 0.34957 0.702006 0.77937C0.702006 0.991404 0.776505 1.16905 0.925502 1.31805L5.24069 5.73066C5.42407 5.91977 5.61891 6 5.85387 6Z"
+                fill="#888888"
+              />
+            </svg>
```
Comment (kylezryr) [inline components/TipDropdown/index.tsx]: these icons should be defined in the `lib/icons` file and imported here instead!

File: components/TipDropdown/styles.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import styled from 'styled-components';
+import Icon from '../Icon';
+
+export const OrderedTipList = styled.ol`
+  font-size: 12px;
+  font-weight: 300;
+  padding: 1.5rem;
+  margin: 1rem;
+  margin-top: -0.5rem;
+  margin-bottom: -0.5rem;
```
Comment (kylezryr) [inline components/TipDropdown/styles.ts]: negative margins 😱 is there any way we can avoid using them?

File: app/resources/page.tsx [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+    const [fullList, setFullList] = useState<PlantTip[]>([]);
+    useEffect(() => {
+      (async () => {
+        const tipList = await getAllPlantTips();
+        setFullList(tipList);
+      })();
+    });
+    return (
+      <>
+        <Box $ml="1.5rem" $mb=".5rem">
```
Comment (kylezryr) [inline app/resources/page.tsx]: we need to make the `ml` a `pl` instead because the margin makes the box move off the screen and causes there to be whitespace to the right (you can figure this out by inspecting element, going into the 'inspect mode' and hovering over each of the components to see their spacing and borders and stuff)
image

**Review by kylezryr — CHANGES_REQUESTED** [review]
great work overall! couple of changes to be made but works fine n all of that! regarding the coloring of the icons, i'd just import new ones and name them something different tbh to make sure they're the right color

File: app/view-plants/page.tsx
Comment (torinnayak1) [reply app/view-plants/page.tsx:1]: oh shoot im so sorry i thought i made a new branch but I must not have, sorry

File: components/TipDropdown/styles.ts (hunk 8 above)
Comment (torinnayak1) [reply components/TipDropdown/styles.ts]: yeah, i think i was probably being lazy, not sure tho

**Review by kylezryr — CHANGES_REQUESTED** [review]
changes look good! only thing is to import the correctly colored icons to use for water management and planting tips, then we're ready to merge!

**please run git pull before you make changes, I rebased your branch!**

File: components/TipDropdown/index.tsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+export function TipDropdown({ name, tips, icon }: TipDropdownProps) {
+  const [expanded, setExpanded] = useState(false);
+  return (
+    <Flex
+      $direction="column"
+      $justify="center"
+      $m="1.5rem"
+      $mb="-.5rem"
```
Comment (kaleenchen) [inline components/TipDropdown/index.tsx]: woah didn't know we can have neg margins

**Review by kaleenchen — APPROVED** [review]
looks good yay!!

**Review by kylezryr — APPROVED** [review]
ready 2 merge
